### DISCUSSION TITLE
Unicode support for event names

### DISCUSSION THREAD

**Discussion by zombie6888** [invertase/react-native-firebase#4523] [Ideas]
Hello, we are using firebase analytics, but library doesn't support different languages for event names. Current validation fail with unicode characters because of this regular expression /^[a-zA-Z0-9_]+$/;

https://github.com/invertase/react-native-firebase/blob/master/packages/app/lib/common/validate.js#L20

In our project we added hack to avoid checking:

`RegExp.prototype.test = () => retun true`

It removes validation before event sending and it works! I can see event names on russian language 
in firebase console.

So, is it possible to improve validation for library to support unicode characters?

**Comment by mikehardy** [comment]
Sure! If you can reference the upstream documentation that specifies exactly what characters are acceptable (I expect the unicode range, yes?) then implement a regex that handles the range the upstream doc specifies, we can merge it. There is a similar PR in progress now so you can see similar work #4522

**Comment by zombie6888** [comment]
I found only this docs: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Event
with reference to this method: 
https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#isLetterOrDigit-int-
